server/test/integration/com/thoughtworks/go/config/GoConfigMigrationIntegrationTest.java

```diff
@@ -1115,6 +1116,73 @@ public void shouldTrimLeadingAndTrailingWhitespaceFromCommandsInTemplates_asPart
         assertThat((ExecTask) task, is(new ExecTask("c:\\program files\\cmd.exe", "arguments", (String) null)));
     }
 
+    @Test
+    public void shouldNotRemoveNonEmptyUserTags_asPartOfMigration77() throws Exception {
+        String configXml =
+                "<cruise schemaVersion='76'>"
+                +"  <pipelines group='first'>"
+                +"    <authorization>"
+                +"       <view>"
+                +"         <user>abc</user>"
+                +"       </view>"
+                +"    </authorization>"
+                +"    <pipeline name='Test' template='test_template'>"
+                +"      <materials>"
+                +"        <hg url='../manual-testing/ant_hg/dummy' />"
+                +"      </materials>"
+                +"     </pipeline>"
+                +"  </pipelines>"
+                +"</cruise>";
+        String migratedXml = migrateXmlString(configXml, 76);
+        assertThat(migratedXml, containsString("user"));
+    }
+
+    @Test
+    public void shouldRemoveEmptyTags_asPartOfMigration77() throws Exception {
+        String configXml =
+                "<cruise schemaVersion='76'>"
+                +"  <pipelines group='first'>"
+                +"    <authorization>"
+                +"       <view>"
+                +"         <user>foo</user>"
+                +"         <user></user>"
+                +"         <user>      </user>"
+                +"       </view>"
+                +"    </authorization>"
+                +"    <pipeline name='Test' template='test_template'>"
+                +"      <materials>"
+                +"        <hg url='../manual-testing/ant_hg/dummy' />"
+                +"      </materials>"
+                +"     </pipeline>"
+                +"  </pipelines>"
+                +"</cruise>";
+        String migratedXml = migrateXmlString(configXml, 76);
+        assertThat(StringUtils.countMatches(migratedXml, "user"), is(2));
```

Comment:
Should this not be `1` user? I'd think `<user>    </user>` should be treated the same as `<user`>`